README.md

```diff
@@ -139,26 +139,129 @@ Dedicated servers hosted on Steam are usually required to be running the *latest
 
 While the use of build cache can help drastically reduce update durations, it cannot be utilized as the game images are built for public use, purposefully done so using public machines.
 
-## Build history
+### Build history
 
 The project uses multiple CI services for its build jobs. You can find the history of past build jobs by clicking on their corresponding build status badges.
+
+## Usage
+
+### Docker
+
+The project assumes knowledge concerning the [`docker`](https://docs.docker.com/) runtime. Instructions on customization and orchestration of containerized game server instances are outside the scope the project.
+
+#### Entrypoint and CMD
+
+Currently, the default **ENTRYPOINT** for all game images is [`"bash", "-c"`](build/Dockerfile#L72), and the **CMD** is [`""`](build/Dockerfile#L73). This makes it convenient especially in development environments in that the game's command line can simply be appended as the final argument of the `docker run` command.
+
+Each of the values can be overridden at runtime which is well supported by container orchestration tools such as [Kubernetes](https://kubernetes.io/) and  [Docker Swarm Mode](https://docs.docker.com/engine/swarm/), and the standalone tool, [Docker Compose](https://docs.docker.com/compose/). Alternatively, they can be modified as part of the build steps in custom images.
+
+#### Workdir
+
+The default working directory for all the images is [`/server`](build/Dockerfile#L70) within which all of a game's dedicated server files reside.
+
+#### Command line
+
+The following are some examples of how the game servers can be run:
+
+```shell
+# Counter-Strike: Global Offensive
+docker run -it -p 27015:27015/udp sourceservers/csgo:latest 'srcds_linux -game csgo -port 27015 +game_type 0 +game_mode 1 +mapgroup mg_active +map de_dust2'
+
+# Counter-Strike 1.6
+docker run -it -p 27016:27016/udp goldsourceservers/cstrike:latest 'hlds_linux -game cstrike +port 27016 +maxplayers 10 +map de_dust2'
+```
+
+* `-t` for a pseudo-TTY is mandatory; servers may not run correctly without it
+* `-i` for STDIN for interactive use of the game console
+* `-d` for running the container in detached mode
+
+#### Attaching
+
+If the game process is running as `PID 1` and STDIN is enabled for the container, the game's console can be accessed via:
+
+```shell
+docker attach containername
+```
+
+#### Debugging
+
+To debug a container or its files:
+
+```shell
+# To enter into a container
+docker exec -it containername bash
+
+# To issue detached command(s)
+docker exec containername ps aux                                     # Single, simple command
+docker exec containername bash -c 'printenv && ls -al && ps aux'     # Multiple or advanced commands
+```
+
```

Comment:
```suggestion

## Updating gameservers

To update a gameserver, pull the `:latest` docker image again, and then restart the server.

```sh
docker pull sourceservers/csgo:latest 
docker rm -f csgo-server 
docker run --name csgo-server -it -p 27015:27015/udp sourceservers/csgo:latest srcds_linux -game csgo -port 27015 +game_type 0 +game_mode 1 +mapgroup mg_active +map de_dust2 
```

There are many ways to detect when a gameserver needs an update, but this is out of the scope of this repository. However here is an [example](https://stackoverflow.com/a/44740429/3891117) of a cronjob you may use to update your server.

Reply:
Update the `docker run` command to quote the server command line.